controllers/v1/reportsController.js

```diff
@@ -3012,4 +3017,238 @@ module.exports = class Reports {
+    try {
+      if (!req.query.submissionId) {
+        let response = {
+          result: false,
+          message: 'submissionId is a required field',
+        };
+        res.send(response);
+      } else {
+        let submissionId = req.query.submissionId;
+
+        let surveySubmissionsDocumentArray = await surveySubmissionsHelper.surveySubmissionDocuments({
+          _id: submissionId,
+          status: 'completed',
+        });
+
+        let surveySubmissionsDocument = surveySubmissionsDocumentArray[0];
+
+        if (!surveySubmissionsDocument) {
+          throw { message: messageConstants.apiResponses.SUBMISSION_NOT_FOUND };
+        }
+
+        //adding question options, externalId to answers array
+        if (surveySubmissionsDocument.answers && Object.keys(surveySubmissionsDocument.answers).length > 0) {
+          surveySubmissionsDocument = await questionsHelper.addOptionsToSubmission(surveySubmissionsDocument);
+        }
+
+        let solutionDocument = await solutionsHelper.solutionDocuments(
+          {
+            _id: surveySubmissionsDocument.solutionId,
+          },
+          ['name', 'scoringSystem', 'description', 'questionSequenceByEcm']
+        );
+
+        if (!solutionDocument.length) {
+          throw messageConstants.apiResponses.SOLUTION_NOT_FOUND;
+        }
+
+        solutionDocument = solutionDocument[0];
+        surveySubmissionsDocument['solutionInfo'] = solutionDocument;
+
+        if (surveySubmissionsDocument.programId && surveySubmissionsDocument.programId != '') {
+          let programDocument = await programsHelper.list(
+            {
+              _id: surveySubmissionsDocument.programId,
+            },
+            ['name', 'description']
+          );
+
+          if (!programDocument[0]) {
+            throw messageConstants.apiResponses.PROGRAM_NOT_FOUND;
+          }
+          surveySubmissionsDocument['programInfo'] = programDocument[0];
+        }
+        let data = surveySubmissionsDocument;
+
+        let transformedData = transformData(data);
+
+        let chartData = await helperFunc.instanceReportChart(transformedData, filesHelper.survey);
+        chartData.solutionName = data.solutionExternalId;
+
+        let surveyAnswers = data.answers;
+
+        let evidenceData = formateEvidenceData(surveyAnswers, chartData);
+
+        let responseObj;
+
+        if (evidenceData.length > 0) {
+          responseObj = await helperFunc.evidenceChartObjectCreation(
+            chartData,
+            evidenceData
+          );
+        } else {
+          responseObj = chartData;
+        }
+
+        return resolve({
+          status: httpStatusCode.ok.status,
+          message: responseObj.response,
+        });
+      }
+    } catch (err) {
+
+      let response = {
+        result: false,
+        message: err.message || 'INTERNAL_SERVER_ERROR',
+      };
+      return resolve(response);
+    }
+  });
+};
+
 };
+
+function transformData(data) {
```

Comment:
@borkarsaish65 function signature and description is missing. Please add

Reply:
I have added now @VISHNUDAS-tunerlabs

Reply:
rewrote entire logic